upi/vsphere/host_a_record/variables.tf

```
@@ -0,0 +1,16 @@
+variable "zone_id" {
+  type        = string
+  description = "The ID of the hosted zone to contain this record."
+}
+
+variable "host_type" {
+  type = string
+}
+variable "cluster_domain" {
+  type = string
+}
+
+variable "ip_addresses" {
+  type = list(string)
+}
+
```

I think this should be 
```suggestion
variable "records" {
  type = map(string)
  descriptions = "A records to be added to the zone_id"
}
```

```
ip_addresses
hostnames
// https://www.terraform.io/docs/configuration/functions/zipmap.html
records = zipmap(hostnames, ip_addresses)
```